src/main/static/src/components/henkilo/OppijaViewContainer.js

```diff
@@ -9,30 +11,44 @@ import {updateHenkiloNavigation} from "../../actions/navigation.actions";
 import {oppijaNavi} from "../navigation/navigationconfigurations";
 import PropertySingleton from '../../globals/PropertySingleton'
 import HenkiloViewPage from "./HenkiloViewPage";
+import type {Navigation} from "../../actions/navigation.actions";
+import type {Tab} from "../../types/tab.types";
+import type {HenkiloState} from "../../reducers/henkilo.reducer";
+import type {L10n} from "../../types/localisation.type";
+import type {Locale} from "../../types/locale.type";
+import type {KoodistoState} from "../../reducers/koodisto.reducer";
+
+type Props = {
+    oidHenkilo: string,
+    henkilo: HenkiloState,
+    fetchHenkiloSlaves: (oid: string) => void,
```

Comment:
Tavallaan pitäisin enemmän siitä, että komponentit hakevat reduxista itse tarvitsemansa toiminnot ja ylemmän tason komponentit eivät oksentaisi ihan kaikkea mahdollista alemman tason komponenteille. Mutta ei ehkä ole tämän taskin pääasiallinen vaatimus.

Reply:
Joo, no tässä ei oo selkeää käytäntöä. Ymmärtääkseni connectin käyttämistä pitäisi enemmän vältellä kuin käyttää aina mahdollisuuksien tullen. Mutta en toki toistaiseksi oikein keksi muuta selkeästi ikävää puolta kuin sen, että ide ei osaa näyttää missä komponenttia käytetään, kun se wräpätään connectin sisään. 

Joka tapauksessa tämä olisi the uusi oikea tapa https://reactjs.org/docs/context.html :)

Reply:
Alunperin reduxin dokumentaatiossahan sanottiin, että connect() pitäisi käyttää vain ylätason komponenteissa minkä takia tätä on seurattu useissa projekteissa. Nykyään dokumentaatioon on päivitetty, että connect() on parasta käyttää kaikissa komponenteissa.